### ISSUE TITLE
View.refresh()

### ISSUE THREAD

**Issue by SephReed** [NativeScript/NativeScript#7121]
**Is your feature request related to a problem? Please describe.**
 
I have a view which is being rendered incorrectly.  If I run the code:
```
const className = view.className;
view.className = "";
setTimeout(() => view.className = className, 500);
```
The view renders correctly, after the timeout. 

**Describe the solution you'd like**

I'd like to be able to rerender with a single function call.  Something like View.refresh()

**Describe alternatives you've considered**

Alternatively, the same class should produce the same results, but sometimes it doesn't.  This is just a means to keep moving forwards if anything goes wrong consistently with nativescript.

**Additional context**

In case such a function already exists, I've asked about it on stack overflow: https://stackoverflow.com/questions/55578812/how-can-i-force-ui-refresh-in-nativescript-core

**Comment by NickIliev** [comment]
@SephReed chaning CSS classes should be supported out-of-the-box. Can you elaborate more on your exact the scenario? I've tried to reproduce the issue but to no avail - [here is the test Playground](https://play.nativescript.org/?template=play-tsc&id=e6wtdj) In which I am removing the CSS by resetting the className just as in your code snippet. Everything seems to works as expected n this cae.

Please, if possible, provide a sample Playground that can reproduce the case;

**Comment by SephReed** [comment]
I've narrowed it down to IOS, if a view is added, immediately has translateX set to -screenWidth, then animates to x = 0. (as in is added off page and slides in), then the styling will be broken and nothing but resetting the css (with delay) will fix it.

I'm not sure I can make an example, but I also don't think it's necessary.  Sometimes things break, and it would be nice to have a backup function to sort of force them to work again.  That being said, requestLayout() is apparently supposed to do that, so I think this is another issue.

**Comment by SephReed** [comment]
Alright, I've really narrowed it down now:

On iOS, if you use `StackLayout` and have the `horizontal-align` css property set to center, for a view that is rendered with non-zero translateX or Y... something breaks and cascades into the child elments.

What is currently working for me is to not use the `horizontal-align` css property at all, but instead use flex -> align-items.

**Comment by SephReed** [comment]
Hmph.  Or I thought I had it.

Anything having to do with flex or centering is broken when rendering happens on an item translated off the screen _to the left_.  What's very interesting is that the same screens when animated in from the right are fine.  It is only when they come from the left.  Totally peculiar.

**Comment by SephReed** [comment]
And I think I finally got it.  Setting translateX to a negative value before a view is rendered on ios will break the views layout.  Putting a timeout of 5ms between adding the view, and translating it off screen has been enough to keep it from happening for me.

**Comment by NickIliev** [comment]
@SephReedtry to wrap all the translation logic in `isLoaded` as you have done for your consequential animation logic.

```JS
// instead of setTimeout
if (addMe.isLoaded) {
```

The `requestLayout` should do exactly what need (triggering a new layout pass to re-render all elements) -is that not working for your scenario? Basically, all scenarios where the UI should be remeasured and re-rendered should be handled by design, so this is the reason I want to investigate the case with a sample application or Playground. If there is a case where the UI is broken after applying CSS I would consider this one a bug and not a feature for `refresh` method.

**Comment by SephReed** [comment]
It's definitely a bug, and the feature request should have (by my understanding) already been fulfilled by `requestLayout()`.   

> try to wrap all the translation logic in isLoaded as you have done for your consequential animation logic.

I did.  So far, I've run into multiple issues involved with modifying views too early on iOS, and the only "solution" is setTimeout().  Waiting for load is not sufficient.  That being said, I know just how precarious setTimeout() solutions are, and really would like something better.

Does anything akin to `addEventListener(View.nativeViewLoadedEvent, () => ...)` exist?  On load seems to have more to do with nativescript than the actual native views (at least on ios).

**Comment by NickIliev** [comment]
@SephReed going back to your initial request

```TSC
const className = view.className;
view.className = "";
setTimeout(() => view.className = className, 500);
```

Try to isolate this case ina  Playground app.  As I stated before, any change in the CSS (as in the snippet above) should be handled by design so if there is a case where this is not happening I will mark the issue as a bug (this is not a feature request if the CSS is not updated). However, I am suspecting that your logic is trying to do additional things on non-rendered views (e.g. animations, translation) which could manifest different issue.

So basically a test app is a must to investigate further.

**Comment by SephReed** [comment]
> However, I am suspecting that your logic is trying to do additional things on non-rendered views (e.g. animations, translation) which could manifest different issue.

Yeah, this was the issue.  I've since moved to using timeouts to try to make sure all of my views are rendered before acting on them.

I looked into whether or not an `onViewRendered()` function could be created, but judging by this answer on Stack Overflow, a timeout is about as good as it gets:

https://stackoverflow.com/questions/42420178/how-to-simply-wait-for-any-layout-in-ios

Still, all of this being said, a catch-all-force-rerender function would be nice to have, if not maybe a bit easy to abuse.
